unit_tests/test_simulation_info.py

```diff
@@ -0,0 +1,60 @@
+import unittest
+import os
+import inspect
+import d3a_api_client
+
+from d3a_api_client.utils import get_simulation_info, domain_name_from_env, \
+    websocket_domain_name_from_env, simulation_id_from_env, SimulationInfoException
+from cli import set_simulation_file_env
+
+
+class TestSimulationInfo(unittest.TestCase):
+
+    def setUp(self) -> None:
+        os.environ['JSON_FILE_PATH'] = ""
+        self.api_client_path = os.path.dirname(inspect.getsourcefile(d3a_api_client))
+
+    def tearDown(self) -> None:
+        os.environ['JSON_FILE_PATH'] = ""
+
+    def test_simulation_info_returns_the_default_when_no_json_file_path(self):
+        simulation_id, domain_name, websockets_domain_name = get_simulation_info()
+        assert simulation_id == simulation_id_from_env
+        assert domain_name == domain_name_from_env
+        assert websockets_domain_name == websocket_domain_name_from_env
+
+    def test_simulation_info_file_is_correctly_parsed_json_file(self):
+        os.environ['JSON_FILE_PATH'] = os.path.join(
+            self.api_client_path,
+            "resources/api-client-summary-84d221fa-46e1-49e0-823b-26cfb3425a5a.json"
+        )
+        simulation_id, domain_name, websockets_domain_name = get_simulation_info()
+        assert simulation_id == "84d221fa-46e1-49e0-823b-26cfb3425a5a"
+        assert domain_name == "localhost"
+        assert websockets_domain_name == "ws://localhost/external-ws"
+
+    def test_assertion_is_raised_if_incorrect_info_is_provided(self):
+        with self.assertRaises(SimulationInfoException):
+            os.environ['JSON_FILE_PATH'] = os.path.join(
+                self.api_client_path,
+                "resources/malformed-api-client-summary-84d221fa-46e1-49e0-823b-26cfb3425a5a.json"
+            )
+            get_simulation_info()
+
+    def test_default_simulation_info_is_under_setup_module_if_base_path_isnt_provided(self):
+        set_simulation_file_env(base_setup_path=None, simulation_info='test.json',
+                                run_on_redis=False)
+        assert os.environ['JSON_FILE_PATH'] == os.path.join(self.api_client_path,
+                                                            'setups/test.json')
```

Comment:
I thought we agreed on not supporting this: 
https://github.com/gridsingularity/d3a-api-client/pull/94#discussion_r596166482

Reply:
Its not the same as linked comment, here `simulation_info ` two lines above via `set_simulation_file_env ` is set as as `test.json` and thats what I'm testing if found in `setups` folder (being the default base location of api-client scripts), whereas what you said in the linked comment(which I agreed) was to not simply pick the first `.json` being found in targeted directory.

Reply:
Indeed, sorry, I misunderstood here.